packages/walletconnect-v2/src/index.ts

```diff
@@ -86,27 +102,54 @@ export class WalletConnect extends Connector {
     this.events.emit(URI_AVAILABLE, uri)
   }
 
+  private async initializeProvider(
+    desiredChainId: number | undefined = this.defaultChainId
+  ): Promise<WalletConnectProvider> {
+    const ethProviderModule = await import('@walletconnect/ethereum-provider')
+    const rpcMap = this.rpcMap ? await getBestUrlMap(this.rpcMap, this.timeout) : undefined
+    const chainProps = this.getChainProps(desiredChainId)
+
+    this.provider = await ethProviderModule.default.init({
+      ...this.options,
+      ...chainProps,
+      rpcMap,
+    })
+
+    return this.handleProviderEvents(this.provider)
+  }
+
+  // Helper function to reorder chains array based on desiredChainId
+  private reorderChainsBasedOnId(
+    chains: number[] | undefined,
+    desiredChainId: number | undefined
+  ): number[] | undefined {
+    if (!chains || !desiredChainId || !chains.includes(desiredChainId) || chains.length === 0) {
```

Comment:
@JFrankfurt 

This specifically

>  !chains.includes(desiredChainId) 

is what is causing the tests to fail. 

You should not handle this case here. If you take a look at `getChainsWithDefault` code, you will see it handles that case on its own:

https://github.com/Uniswap/web3-react/blob/505e164e156854e7a82bc56520ae0fab3129746d/packages/walletconnect-v2/src/utils.ts#L88-L94